artiq/browser/experiments.py

```diff
@@ -341,7 +343,7 @@ async def _get_run_task(self, expid):
         logger.info("Running '%s'...", self.expurl)
         worker = Worker(self._area.worker_handlers)
         try:
-            await worker.build(rid=None, pipeline_name="browser",
+            await worker.build(rid=self.rid, pipeline_name="browser",
```

Comment:
`rid` cannot be `None` as it will fail further at formatting https://github.com/m-labs/artiq/blob/8740ec3dd52d85084237797881ea137492bfe070/artiq/master/worker_impl.py#L289